ext/datastore/bigquery/external_table_spec.go

```diff
@@ -0,0 +1,71 @@
+package bigquery
+
+import (
+	"fmt"
+
+	"github.com/odpf/optimus/models"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+const (
+	ExternalTableTypeGoogleSheets ExternalTableType = "GOOGLE_SHEETS"
+)
+
+type ExternalTableType string
+
+// BQExternalSource specifies table source information for external data source
+type BQExternalSource struct {
+	SourceType string `yaml:"type,omitempty" structs:"type"`
+
+	// External Table URI string for the referenced spreadsheets
+	SourceURIs []string `yaml:"uris,omitempty" structs:"uris,omitempty"`
+
+	// Additional configs for CSV, GoogleSheets, Bigtable, and Parquet formats.
+	Config map[string]interface{} `yaml:"config,omitempty" structs:"config"`
+}
+
+type externalTableSpec struct{}
+
+func (s externalTableSpec) Adapter() models.DatastoreSpecAdapter {
+	return &tableSpecHandler{}
+}
+
+func (s externalTableSpec) Validator() models.DatastoreSpecValidator {
+	return func(spec models.ResourceSpec) error {
+		if !tableNameParseRegex.MatchString(spec.Name) {
+			return fmt.Errorf("for example 'project_name.dataset_name.table_name'")
+		}
+		parsedNames := tableNameParseRegex.FindStringSubmatch(spec.Name)
+		if len(parsedNames) < 3 || len(parsedNames[1]) == 0 || len(parsedNames[2]) == 0 || len(parsedNames[3]) == 0 {
+			return fmt.Errorf("for example 'project_name.dataset_name.table_name'")
+		}
+		return nil
+	}
+}
+
+func (s externalTableSpec) DefaultAssets() map[string]string {
+	return map[string]string{}
+}
+
+func extractTableSourceFromProtoStruct(protoVal *structpb.Value) *BQExternalSource {
+	sInfo := &BQExternalSource{}
+	if protoVal.GetStructValue() == nil {
+		return sInfo
+	}
+	if f, ok := protoVal.GetStructValue().Fields["type"]; ok {
+		sInfo.SourceType = f.GetStringValue()
```

Comment:
don't we need to uppercase the value to match with our type in case it's not already?

Reply:
Agreed that such validation should be there. Will add